Desktop/data/datasettablemodel.h

```diff
@@ -45,11 +45,19 @@ class DataSetTableModel : public DataSetTableProxy
 
 	columnType				getColumnType(size_t column)			const				{ return DataSetPackage::pkg()->getColumnType(column);								}
 	std::string				getColumnName(size_t col)				const				{ return DataSetPackage::pkg()->getColumnName(col);									}
+	int						getColumnIndex(const std::string& col)	const				{ return DataSetPackage::pkg()->getColumnIndex(col);								}
+	QStringList				getColumnLabelsAsStringList(int col)	const				{ return DataSetPackage::pkg()->getColumnLabelsAsStringList(col);					}
+	size_t					getMaximumColumnWidthInCharacters(int index) const			{ return DataSetPackage::pkg()->getMaximumColumnWidthInCharacters(index);			}
+	QModelIndex				parentModelForType(parIdxType type, int column = 0)	const	{ return DataSetPackage::pkg()->parentModelForType(type, column);					}
+
 				bool		showInactive()							const				{ return _showInactive;	}
 
 signals:
 				void		columnsFilteredCountChanged();
 				void		showInactiveChanged(bool showInactive);
+				void		columnTypeChanged(QString colName);
```

Comment:
Or maybe you left it at the `resetModel` style instead of fixing `dataChanged`?

Reply:
I don't understand... with columnTypeChanged I have precisely the information I need to change the type. I don't hae to refresh all models.

Reply:
Well, to have DataSetPackage as the root of all models all changes should ideally go through that. But I see that it comes from there so it is fine.